node_agent/main.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017 Istio Authors
```

Comment:
Actually, the existing directory structure is unconventional. The cmd/ looks like cobra command directory. But from the cobra's documentation, the intended behavior is to have the cmd/ under the app directory. For example the directory structure would have: 

istio_ca
   cmd/
   ... any other dirs...
   BUILD
   main.go

For example the controller/secret.go is completely related to istio_ca and can be moved into the istio_ca directory. The higher level directory should only have the code that is common ACROSS istio_ca and node_agent. This is why I have a utils directory with tls.go since this is going to be used by Istio CA also. 

This is the structure I followed for node_agent. This will also keep all the related files together rather than scattered in multiple places. I think this keeps the directory clean. I think we must revisit the directory structure.